homeassistant/components/switcher_kis/sensor.py

```diff
@@ -0,0 +1,131 @@
+
+
+POWER_SENSORS = {
+    "power_consumption": AttributeDescription(
+        name="Power Consumption",
+        unit=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+        default_value=0,
+    ),
+    "electric_current": AttributeDescription(
+        name="Electric Current",
+        unit=ELECTRICAL_CURRENT_AMPERE,
+        device_class=DEVICE_CLASS_CURRENT,
+        state_class=STATE_CLASS_MEASUREMENT,
+        default_value=0.0,
+    ),
+}
+
+TIME_SENSORS = {
+    "remaining_time": AttributeDescription(
+        name="Remaining Time",
+        icon="mdi:av-timer",
+        default_value="00:00:00",
+    ),
+    "auto_off_set": AttributeDescription(
+        name="Auto Shutdown",
+        icon="mdi:progress-clock",
+        default_enabled=False,
+        default_value="00:00:00",
+    ),
+}
+
+SENSORS = {**POWER_SENSORS, **TIME_SENSORS}
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: dict,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType,
+) -> None:
+    """Set up Switcher sensor from config entry."""
+    device_data = hass.data[DOMAIN][DATA_DEVICE]
+
+    async_add_entities(
+        SwitcherSensorEntity(device_data, attribute, SENSORS[attribute])
+        for attribute in SENSORS
+    )
+
+
+class SwitcherSensorEntity(SensorEntity):
+    """Representation of a Switcher sensor entity."""
+
+    def __init__(
+        self,
+        device_data: SwitcherV2Device,
+        attribute: str,
+        description: AttributeDescription,
+    ) -> None:
+        """Initialize the entity."""
+        self._device_data = device_data
+        self.attribute = attribute
+        self.description = description
+
+        # Entity class attributes
+        self._attr_name = f"{self._device_data.name} {self.description.name}"
+        self._attr_icon = self.description.icon
+        self._attr_unit_of_measurement = self.description.unit
+        self._attr_device_class = self.description.device_class
+        self._attr_entity_registry_enabled_default = self.description.default_enabled
+        self._attr_should_poll = False
+
+        self._attr_unique_id = f"{self._device_data.device_id}-{self._device_data.mac_addr}-{self.attribute}"
+
+    @property
+    def state(self) -> StateType:
+        """Return value of sensor."""
+        value = getattr(self._device_data, self.attribute)
+        if value and value is not WAITING_TEXT:
+            return value
+
+        return self.description.default_value
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, SIGNAL_SWITCHER_DEVICE_UPDATE, self.async_update_data
+            )
+        )
+
+    async def async_update_data(self, device_data: SwitcherV2Device) -> None:
+        """Update the entity data."""
+        if device_data:
```

Comment:
This check doesn't match with the type annotation of the parameter.

Reply:
Looking at the code, this check is redundant, the device data is already checked before the dispatcher is called, I copied it from the switch implementation, but it can't arrive here without data. Will be fixed in a separate PR.